src/Fields/abstract_field.jl

```diff
@@ -168,81 +182,112 @@ total_size(f::AbstractField) = total_size(location(f), f.grid)
 ##### getindex
 #####
 
-@propagate_inbounds Base.getindex(f::AbstractField, inds...) = @inbounds getindex(f.data, inds...)
+# Don't use axes(f) to checkbounds; use axes(f.data)
+Base.checkbounds(f::AbstractField, I...) = Base.checkbounds(f.data, I...)
+
+#=
+# Get this to work...
+parentaxes(f::AbstractField) = ...
+
+function Base.checkbounds(::Type{Bool}, A::AbstractField, I...)
+    @_inline_meta
+    checkbounds_indices(Bool, parentaxes(A), I)
+end
+=#
```

Comment:
Is this still needed? Seems like it works a couple of lines above.

Reply:
Hmm yeah I think this is a fix for the wrong error. I'll get rid of it.